### PR TITLE
Issue#33 first migration

### PR DESCRIPTION
**Pull request by Arthurpanda**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: home-application/src/main/java/com/softserveinc/ita/homeproject/HomeApplication.java [hunk 1]
```diff
@@ -7,5 +7,6 @@
 public class HomeApplication  {
     public static void main(String[] args) {
          SpringApplication.run(HomeApplication.class, args);
+
```
Comment (kh0ma) [inline home-application/src/main/java/com/softserveinc/ita/homeproject/HomeApplication.java]: Why do we need this extra line?

File: home-data-migration/pom.xml [hunk 2]
```diff
@@ -12,5 +12,52 @@
     <packaging>jar</packaging>
 
     <artifactId>home-data-migration</artifactId>
+    <dependencies>
+        <!-- https://mvnrepository.com/artifact/org.liquibase/liquibase-core -->
 
+        <dependency>
+            <groupId>org.liquibase</groupId>
+            <artifactId>liquibase-core</artifactId>
+            <version>4.0.0</version>
```
Comment (kh0ma) [inline home-data-migration/pom.xml:19]: Can we introduce a property?

File: home-data-migration/pom.xml [hunk 3]
```diff
@@ -12,5 +12,52 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.postgresql</groupId>
+            <artifactId>postgresql</artifactId>
+            <scope>runtime</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework</groupId>
```
Comment (kh0ma) [inline home-data-migration/pom.xml:22]: ?

File: home-data-migration/pom.xml [hunk 4]
```diff
@@ -12,5 +12,52 @@
...
+            <artifactId>spring-jdbc</artifactId>
+        </dependency>
+    </dependencies>
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.liquibase</groupId>
+                <artifactId>liquibase-maven-plugin</artifactId>
+                <version>3.6.3</version>
```
Comment (kh0ma) [inline home-data-migration/pom.xml:36]: use property

File: home-data-migration/pom.xml [hunk 5]
```diff
@@ -12,5 +12,52 @@
...
+                    <url>${db_HOST}</url>
+                    <username>${db_USERNAME}</username>
+                    <password>${db_PASSWORD}</password>
+                    <driver>org.postgresql.Driver</driver>
+                </configuration>
+                <dependencies>
+                </dependencies>
+                <executions>
+                    <execution>
+                        <id>202011092145</id>
```
Comment (kh0ma) [inline home-data-migration/pom.xml]: What is the magic number? Can you properly name this execution?

File: home-data-migration/src/main/java/Main.java [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import java.util.Scanner;
+
+public class Main {
+    public static void main(String[] args) {
+        String pass, user;
+        Scanner scanner = new Scanner(System.in);
+        System.out.println("Please, enter the username");
+        user = scanner.nextLine();
```
Comment (kh0ma) [inline home-data-migration/src/main/java/Main.java]: Please do not use any prompts. We'll use this app in a non-tty environment so just take required data from arguments.

File: home-data-migration/src/main/java/LiquibaseUpdate.java [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+import java.sql.SQLException;
+import java.util.Properties;
+
+public class LiquibaseUpdate {
+//    name of references in migration.properties
+    private static final String dbhost = "url";
+    private static final String pathname = "changelog.path";
+    private static final String changelog = "changelog.name";
+
+    public void runLiquibase(String user, String password) {
```
Comment (kh0ma) [inline home-data-migration/src/main/java/LiquibaseUpdate.java]: With a high probability URL of db will vary on the environments. Please use the default if not present approach and give a possibility to specify it.

File: home-data-migration/src/main/java/LiquibaseUpdate.java [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+                } catch (SQLException e) {
+                    //nothing to do
+                }
+            }
+        }
+    }
+//    Get value from migration.properties
+    public String getFromProperties(String value) {
+        String url = "";
+        try (InputStream input = new FileInputStream("home-data-migration\\src\\main\\resources\\migration.properties")) {
```
Comment (kh0ma) [inline home-data-migration/src/main/java/LiquibaseUpdate.java]: I don't think this URI will be present somewhere out of the project. Please specify one from the classpath.

File: home-data-migration/src/main/java/LiquibaseUpdate.java [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+        Liquibase liquibase;
+        try {
+            Class.forName("org.postgresql.Driver");
+            connection = DriverManager.getConnection(url, user, password);
+            Database database = DatabaseFactory.getInstance().
+                    findCorrectDatabaseImplementation(new JdbcConnection(connection));
+            liquibase = new Liquibase(changelogName, new FileSystemResourceAccessor(new File(path)), database);
+            liquibase.update(new Contexts());
+        } catch (SQLException | LiquibaseException | ClassNotFoundException e) {
+            e.printStackTrace();
```
Comment (kh0ma) [inline home-data-migration/src/main/java/LiquibaseUpdate.java]: Please wrap it as runtime and rethrow or just declare in the method signature.

File: home-data-migration/src/main/java/LiquibaseUpdate.java [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+        } finally {
+            if (connection != null) {
+                try {
+                    connection.rollback();
+                    connection.close();
+                } catch (SQLException e) {
+                    //nothing to do
```
Comment (kh0ma) [inline home-data-migration/src/main/java/LiquibaseUpdate.java]: I think this condition should be logged as error. Please add logging.

File: home-data-migration/src/main/java/Main.java [hunk 11]
```diff
@@ -0,0 +1,16 @@
+import java.util.Scanner;
+
+public class Main {
```
Comment (kh0ma) [inline home-data-migration/src/main/java/Main.java]: Specify proper name for the class. Use proper package for the class.

File: home-data-migration/src/main/resources/db/changelog/v-1.0/001-changeset-users-table.xml [hunk 12]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+                      http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd">
+
+    <changeSet id="001" author="Artur">
+        <sqlFile dbms="postgresql"
```
Comment (kh0ma) [inline home-data-migration/src/main/resources/db/changelog/v-1.0/001-changeset-users-table.xml:9]: Please get rid of sql file. Use liquibase dsl.

File: home-data-migration/src/main/resources/db/changelog/v-1.0/db.changelog-v.1.0.xml [hunk 13]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+                      http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd">
+
+
+    <include file="001-changeset-users-table.xml" relativeToChangelogFile="true"/>
+<!--    <include file="002-changeset-house-table.xml" relativeToChangelogFile="true"/>-->
```
Comment (kh0ma) [inline home-data-migration/src/main/resources/db/changelog/v-1.0/db.changelog-v.1.0.xml]: Why is it commented out and present in the review without proper comment?

File: home-data-migration/src/main/resources/migration.properties [hunk 14]
```diff
@@ -0,0 +1,4 @@
+url = jdbc:postgresql://localhost:5432/test
+changelog.path = home-data-migration\\src\\main\\resources\\db\\changelog
```
Comment (kh0ma) [inline home-data-migration/src/main/resources/migration.properties]: I think this path isn't relevant for using this app outside of the project sources. Please use the one from classpath.

File: home-data/pom.xml [hunk 15]
```diff
@@ -18,11 +18,18 @@
...
-            <groupId>org.postgresql</groupId>
-            <artifactId>postgresql</artifactId>
-            <scope>runtime</scope>
-        </dependency>
+                <dependency>
+                    <groupId>org.postgresql</groupId>
+                    <artifactId>postgresql</artifactId>
+                    <scope>runtime</scope>
+                </dependency>
+<!--        <dependency>-->
```
Comment (kh0ma) [inline home-data/pom.xml]: Why is it here?

File: home-data/src/main/resources/application.properties [hunk 16]
```diff
@@ -1,3 +1,16 @@
...
+
+#spring.datasource.url=${dbUrl:jdbc:mysql://localhost:3306/test?createDatabaseIfNotExist=true&serverTimezone=UTC}
+#spring.datasource.username=${dbUsername:root}
+#spring.datasource.password=${dbPass:1234}
+#spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
+
+#spring.liquibase.change-log=classpath:db/changelog/db.changelog-master.xml
+spring.jpa.hibernate.ddl-auto=validate
+spring.jpa.database=POSTGRESQL
+spring.jpa.generate-ddl = true
```
Comment (kh0ma) [inline home-data/src/main/resources/application.properties]: ?

File: home-data/src/main/resources/application.properties [hunk 17]
```diff
@@ -1,3 +1,16 @@
-spring.datasource.url=${dbUrl:jdbc:postgresql://localhost:5432/test?createDatabaseIfNotExist=true}
+spring.datasource.url=${dbUrl:jdbc:postgresql://localhost:5432/test}
 spring.datasource.username=${dbUsername:postgres}
-spring.datasource.password=${dbPass:54321}
+spring.datasource.password=${dbPass:1234}
+spring.datasource.driverClassName=org.postgresql.Driver
+
+#spring.datasource.url=${dbUrl:jdbc:mysql://localhost:3306/test?createDatabaseIfNotExist=true&serverTimezone=UTC}
```
Comment (kh0ma) [inline home-data/src/main/resources/application.properties]: ?

File: pom.xml [hunk 18]
```diff
@@ -28,6 +28,11 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <spring.boot.version>2.3.4.RELEASE</spring.boot.version>
         <maven.compiler.plugin.version>3.8.1</maven.compiler.plugin.version>
+
+        <!-- Password and root name for liquibase-->
+        <db_USERNAME>postgres</db_USERNAME>
```
Comment (kh0ma) [inline pom.xml]: use proper case for property names

File: pom.xml [hunk 19]
```diff
@@ -28,6 +28,11 @@
...
+        <db_PASSWORD>1234</db_PASSWORD>
+        <db_HOST>jdbc:postgresql://localhost:5432/test</db_HOST>
```
Comment (kh0ma) [inline pom.xml]: I think db.host doesn't represent the value of this property.

File: pom.xml [hunk 20]
```diff
@@ -72,6 +77,7 @@
             <plugin>
                 <groupId>org.codehaus.mojo</groupId>
                 <artifactId>flatten-maven-plugin</artifactId>
+                <version>1.2.5</version>
```
Comment (kh0ma) [inline pom.xml:79]: Introduce propertry.

File: .github/PULL_REQUEST_TEMPLATE/develop.md [hunk 21]
```diff
@@ -1,7 +1,7 @@
 dev
 ## JIRA
 
-* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)
+* [com.softserveinc.ita.homeproject.migrations.LiquibaseRunner JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)
```
Comment (mykytam) [inline .github/PULL_REQUEST_TEMPLATE/develop.md:4]: Why do we change PR template here?

File: .github/PULL_REQUEST_TEMPLATE/master.md [hunk 22]
```diff
@@ -1,7 +1,7 @@
 master
 ## JIRA
 
-* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)
+* [com.softserveinc.ita.homeproject.migrations.LiquibaseRunner JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)
```
Comment (mykytam) [inline .github/PULL_REQUEST_TEMPLATE/master.md:4]: Why do we change PR template here?

File: home-data/src/main/resources/application.properties [hunk 23]
```diff
@@ -1,5 +1,3 @@
-spring.datasource.url=${DATASOURCE_URL:jdbc:postgresql://localhost:5433/postgres}
-spring.datasource.username=${DATASOURCE_USER:postgres}
-spring.datasource.password=${DATASOURCE_PASSWORD:password}
-spring.jersey.type=filter
-server.port=${PORT:8080}
+spring.datasource.url=${dbUrl:jdbc:postgresql://localhost:5432/test}
+spring.datasource.username=${dbUsername:postgres}
+spring.datasource.password=${dbPass:54321}
```
Comment (mykytam) [inline home-data/src/main/resources/application.properties:3]: Why did you wrote here other default values?

File: home-data-migration/src/main/resources/db/changelog/v-1.0/001-create-users-table.sql [hunk 24]
```diff
@@ -0,0 +1,12 @@
+CREATE TABLE if not exists public.users (
+                            id int not null  primary key,
+                            first_name varchar (255)  not null,
+                            last_name varchar (255)  not null,
+                            login varchar (255) not null,
+                            password varchar (255)  not null,
+                            phone_number varchar (255) not null,
```
Comment (mykytam) [inline home-data-migration/src/main/resources/db/changelog/v-1.0/001-create-users-table.sql:7]: Do we really have `_phone_number_` in database? Isn't it should be called` contacts`?

**Pull request closed without merging**